### PR TITLE
docs(git-guides): update git guides

### PR DESCRIPTION
**Pull request by anulman**

### CODE REVIEW

File: scm-git-guides.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+chars, <80 preferred) and ought to look like:
+
+```
+<type>(<scope>): <subject>
+```
+Where:
+- `<type>` categorizes the work performed in the commit, and must be one of:
+    - feat;
+    - fix;
+    - hotfix;
```
Comment (jordanyee) [inline scm-git-guides.md]: What is the purpose of the `hotfix` type? Noticed [Contributing to AngularJS](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit) does not include it. Should this not be simply `fix` in the scope of the commit and `hotfix` in the scope of the branch?

Comment (anulman) [inline scm-git-guides.md]: Agreed, will mod

**Comment by aelavell** [comment]
so meta

**Comment by anulman** [comment]
Should probably add helpful link/s for users less familiar with rebasing.

@inconduit found this one; it checks out: http://rypress.com/tutorials/git/rebasing

**Pull request closed without merging**